README.md

```diff
@@ -294,6 +303,12 @@ faster or equal to the given value.
 
 An Array of the `Worker` instances used by this pool.
 
+### Property: `queueLoad` (readonly)
+
+The current ratio of `queueSize` to `maxQueue`. The value is a number between
+`0` and `1`. If `maxQueue` is not specified, the value will always be `1`
+if the `queueSize` is greater than `0`.
```

Comment:
That doesn’t really make sense, mathematically – if there is no queue load, the ratio would always be `0`?

Reply:
mathematically yeah, but in terms of practical implementation this helps ensure that the drain event always happens when the queue empties. The idea is that if there is no `maxQueue`, having *any* tasks in the queue implies queue load.

Reply:
@jasnell That’s still a bit inconsistent though … if the ratio is specified as `.5`, then for `maxQueue === 100` it fires at `50`, for `maxQueue === 1000` it fires at `500`, for `maxQueue === 10000` it fires at `5000`, but for `maxQueue === Infinity` it fires at `0`? That seems inconsistent, not just mathematically but also with (my) intuition.

I’m starting to feel like this is mixing a few concerns here … Piscina is a thread pooling library, and this feature is more or less implementing a streams-like backpressure mechanism. I’m not sure what another good solution would be, but it might be easier to fire an event whenever a task is done? That would give more flexibility to the user anyway, and limit the complexity here.

Reply:
Effectively what this is saying is that the `queueLoadThreshold` is only activated when `maxQueue` is not Infinity, because we cannot reasonably calculate a load properly without a finite `maxQueue`... so `maxQueue == Infinity` effectively disables the threshold and causes the `drain` to emit when it would by default (which is when `queueSize` is zero).

Reply:
Well, yeah, the `'drain'` event in this format should be emitted when the `queueSize` drops down to zero, I’m just saying that `queueLoadThreshold` should still be `0` in that case and not `1`? I‘m not disagreeing about the event timing here, to be clear

Reply:
The key challenge I have with firing an event on every completion is that it becomes very chatty. That said, I'm going through today and running a number of experiments on this to see if having the threshold actually yields a measurable/significant performance improvement under various conditions.

For example, assuming the feeding stream is feeding content into the pool at a constant pace, we end up with the following metrics on the main thread:

The runtime duration of the example:
![image](https://user-images.githubusercontent.com/439929/81431260-ba0d5c80-9115-11ea-862c-195497f6ce56.png)

The high water mark number of active promises sampled once per event loop turn (as tracked by an async hook):
![image](https://user-images.githubusercontent.com/439929/81431315-cd202c80-9115-11ea-8f1c-593ac0811b5f.png)

Main thread heap usage sampled once per event loop turn:
![image](https://user-images.githubusercontent.com/439929/81431295-c5f91e80-9115-11ea-9ab8-6848cd85f05d.png)

In this example, when the queue is being fed at a constant rate and the workload of each task is constant, changing the queue load has zero measurable impact. 

The next analysis I want to do is adding variability to the rate data is fed into the queue.. and after that, adding variability to the runtime of each task.

Reply:
Bottom line is I don't plan on landing this as is unless I can demonstrate that the queue threshold makes a measurable impact :-)

Reply:
Ok, results here are a bit more interesting... In this sample run, I introduced a random 0-10 millsecond delay in the feeding stream... I changed up how the graphs are displaying and added an additional metric tracking the wait times (max, 90th %ile, and 99th %ile)...

![image](https://user-images.githubusercontent.com/439929/81435033-0360aa80-911c-11ea-8939-5dc09bbc54d1.png)

![image](https://user-images.githubusercontent.com/439929/81435059-0bb8e580-911c-11ea-9842-15f72dd9498e.png)

![image](https://user-images.githubusercontent.com/439929/81435080-14112080-911c-11ea-8069-4a694c0af582.png)

![image](https://user-images.githubusercontent.com/439929/81435103-1ecbb580-911c-11ea-9d7c-89019b7968dd.png)

These are all on a 4 core machine, btw...

Reply:
Ok, and the final scenario... maintains the 0-10 ms random latency feeding the stream and adds a 0-100 ms random latency within the workers ... 

![image](https://user-images.githubusercontent.com/439929/81437479-17a6a680-9120-11ea-9190-6cb92a3fac62.png)

In all this, duration is measuring the time to completely queue up all the work and get through the stream... not the time to actually complete the work. With the lower maxQueue numbers, the `'drain'` event is firing more frequently and the feeder stream is being paused/resumed more frequently, but overall it's not having a drastic impact on feed time no matter how often it fires, which is very good to see.

![image](https://user-images.githubusercontent.com/439929/81437500-21300e80-9120-11ea-966e-5ac00146a683.png)

![image](https://user-images.githubusercontent.com/439929/81437515-268d5900-9120-11ea-823a-1cc3d8137f9d.png)

(the heap on the last one is a blip... running that 100-0.0 test again the heap use falls in line with the others)

![image](https://user-images.githubusercontent.com/439929/81437886-b3381700-9120-11ea-9327-25be18d4e2fa.png)

So it's definitely looking like load threshold isn't going to be significant to justify the additional complexity here but this has still been a valuable exercise in exploring the performance of the pool.